index.js

```diff
@@ -1039,6 +1039,78 @@
       [Fn(a, $.Boolean), m(a), m(a)],
       Z.filterM);
 
+  //# takeWhile :: (Foldable f, Monoid (f a)) =>  (a -> Boolean) -> f a -> f a
+  //.
+  //. Returns the first elements for which the predicate returns true.
```

Comment:
This doesn't sound quite right to me for a non-array argument. How about the following?

_Discards the first inner value which does not satisfy the predicate, and all subsequent inner values._